public/components/integrations/components/setup_integration_inputs.tsx

```diff
@@ -205,9 +211,20 @@ export function IntegrationConnectionInputs({
   dataSourceEnabled: boolean;
   dataSourceManagement: DataSourceManagementPluginSetup;
   savedObjectsMDSClient: SavedObjectsStart;
+  handleSelectedDataSourceChange: (
+    dataSourceMDSId: string | undefined,
+    dataSourceMDSLabel: string | undefined
+  ) => void;
   lockConnectionType?: boolean;
 }) {
+  const [dataSourceMDSId, setDataSourceMDSId] = useState<string | undefined>('');
```

Comment:
What's the difference between `undefined` and empty string here?

Reply:
I can make this consistent to be an empty string in all locations

Reply:
> What's the difference between undefined and empty string here?
`' '` represents the local cluster. I guess the undefined is trying to handle the case that when the MDS is not enabled and the ID is undefined, so at that time we just need to handle the request towards local cluster.

> I can make this consistent to be an empty string in all locations
Ideally, we should still handle the `undefined` and `' '` separately. But just wanna make sure we check the MDS flag correctly so that it will not go thru the MDS endpoint when MDS is disabled.